src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null || obj.getClass() != this.getClass()) {
+            return false;
+        }
+        if (obj == this) {
+            return true;
+        }
+        return width.equals(((Rectangle) obj).width)
```

Comment:
better to cast whole object first and then you can compare its fields without casting them